analyzers/src/SonarAnalyzer.CSharp/Extensions/AssignmentExpressionSyntaxExtensions.cs

```diff
@@ -0,0 +1,113 @@
+            }
+            // var (x, y) = (1, 2)
+            else if (DeclarationExpressionSyntaxWrapper.IsInstance(assignment.Left)
+                && (DeclarationExpressionSyntaxWrapper)assignment.Left is { Designation: { } leftDesignation }
+                && ParenthesizedVariableDesignationSyntaxWrapper.IsInstance(leftDesignation)
+                && TupleExpressionSyntaxWrapper.IsInstance(assignment.Right))
+            {
+                var left = (ParenthesizedVariableDesignationSyntaxWrapper)leftDesignation;
+                var right = (TupleExpressionSyntaxWrapper)assignment.Right;
+                var builder = ImmutableArray.CreateBuilder<AssignmentMapping>(initialCapacity: right.Arguments.Count);
+                MapDesignationElements(builder, left, right);
+                return builder.ToImmutableArray();
+            }
+            else
+            {
+                return ImmutableArray.Create(new AssignmentMapping(assignment.Left, assignment.Right));
+            }
+
+            static void MapTupleElements(ImmutableArray<AssignmentMapping>.Builder builder, TupleExpressionSyntaxWrapper left, TupleExpressionSyntaxWrapper right)
+            {
+                var leftEnum = left.Arguments.GetEnumerator();
+                var rightEnum = right.Arguments.GetEnumerator();
+                while (leftEnum.MoveNext() && rightEnum.MoveNext())
+                {
+                    var leftArg = leftEnum.Current;
+                    var rightArg = rightEnum.Current;
+                    if (leftArg is ArgumentSyntax { Expression: { } leftExpression } && TupleExpressionSyntaxWrapper.IsInstance(leftExpression)
+                        && rightArg is ArgumentSyntax { Expression: { } rightExpression } && TupleExpressionSyntaxWrapper.IsInstance(rightExpression))
+                    {
+                        MapTupleElements(builder, (TupleExpressionSyntaxWrapper)leftExpression, (TupleExpressionSyntaxWrapper)rightExpression);
+                    }
+                    else
+                    {
+                        builder.Add(new AssignmentMapping(leftArg.Expression, rightArg.Expression));
+                    }
+                }
+            }
+
+            static void MapDesignationElements(ImmutableArray<AssignmentMapping>.Builder builder, ParenthesizedVariableDesignationSyntaxWrapper left, TupleExpressionSyntaxWrapper right)
+            {
+                var leftEnum = left.Variables.GetEnumerator();
+                var rightEnum = right.Arguments.GetEnumerator();
+                while (leftEnum.MoveNext() && rightEnum.MoveNext())
+                {
+                    var leftVar = leftEnum.Current;
+                    var rightArg = rightEnum.Current;
+                    if (ParenthesizedVariableDesignationSyntaxWrapper.IsInstance(leftVar)
+                        && rightArg is ArgumentSyntax { Expression: { } rightExpression } && TupleExpressionSyntaxWrapper.IsInstance(rightExpression))
```

Comment:
Similar comment here about breaking the condition into more lines and re-using the proposed method from [this ](https://github.com/SonarSource/sonar-dotnet/pull/5756/files#r903596410) comment.